src/AdminPages/AdminController.php

```diff
@@ -0,0 +1,181 @@
+	 * @param SettingsCrudInterface                $settingsCrud
+	 * @param string                               $fieldsCollectionName
+	 * @param NonceInterface                       $nonce
+	 * @param NonceFieldBuilderInterface           $nonceFieldBuilder
+	 */
+	public function __construct(
+		TemplateInterface $adminSettingsPageRenderer,
+		FormFieldsCollectionBuilderInterface $formBuilder,
+		SettingsCrudInterface $settingsCrud,
+		string $fieldsCollectionName,
+		NonceInterface $nonce,
+		NonceFieldBuilderInterface $nonceFieldBuilder
+	) {
+		$this->adminSettingsPageRenderer   = $adminSettingsPageRenderer;
+		$this->formFieldsCollectionBuilder = $formBuilder;
+		$this->fieldsCollectionName        = $fieldsCollectionName;
+		$this->settingsCrud                = $settingsCrud;
+		$this->nonce = $nonce;
+		$this->nonceFieldBuilder = $nonceFieldBuilder;
+	}
+
+	/**
+	 * Initialize hooks listeners.
+	 */
+	public function init(): void
+	{
+		add_action('woocommerce_settings_tabs_array', [$this, 'registerPluginSettingsTab'], 100);
+		add_action('woocommerce_settings_tabs_' . self::PLUGIN_SETTINGS_TAB_SLUG, [$this, 'renderPluginSettingsPage']);
+		add_action('admin_init', [$this, 'saveSettings'], 11);
+	}
+
+	/**
+	 * Add plugin tab to the WC settings tabs.
+	 *
+	 * @param array $tabs Registered WC settings tabs.
+	 *
+	 * @return array WC settings tabs with this plugin tab added.
+	 */
+	public function registerPluginSettingsTab(array $tabs): array
+	{
+		$tabs[self::PLUGIN_SETTINGS_TAB_SLUG] = _x('Mollie Subscriptions', 'Plugin settings tab name', 'woo-ecurring');
+
+		return $tabs;
+	}
+
+	/**
+	 * Render admin settings page content.
+	 */
+	public function renderPluginSettingsPage(): void
+	{
+		try {
+			$form = $this->formFieldsCollectionBuilder->buildFieldsCollection();
+			$formView = $this->formFieldsCollectionBuilder->buildFormFieldsCollectionView();
+			$nonceField = $this->nonceFieldBuilder->buildNonceField($this->nonce);
+			$nonceFieldView = $this->nonceFieldBuilder->buildNonceFieldView();
+			$context = ['view' => $formView, 'form' => $form, 'nonceField' => $nonceField, 'nonceFieldView' => $nonceFieldView];
+			echo $this->adminSettingsPageRenderer->render($context);
+		}catch ( Throwable $exception) {
+			eCurring_WC_Plugin::debug(
+				sprintf(
+					'Failed to render plugin settings form. Exception was caught with code %1$d and message: %2$s',
+					$exception->getCode(),
+					$exception->getMessage()
+				)
+			);
+
+			echo esc_html_x(
+				'Failed to render plugin settings form. Please, check logs for more details.',
+				'Error message if cannot render plugin settings page',
+				'woo-ecurring'
+			);
+		}
+
+	}
+
+	/**
+	 * Save plugin settings on form submit.
+	 */
+	public function saveSettings(): void
+	{
+		$formData = $_POST[$this->fieldsCollectionName] ?? null;
+
+		if($formData === null){
+			return;
+		}
+
+		if(! $this->isNonceValid())
+		{
+			//todo: add admin notice here.
```

Comment:
Just to make sure, can this be done later?

Reply:
I guess, it's better to do now. For now, I just added `wp_nonce_ays()` to print the error message and exit.